pkg/controller/common/resource_reconciler_test.go

```diff
@@ -2,20 +2,34 @@ package common
 
 import (
 	"context"
+	"os"
+	"testing"
+	"time"
+
 	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
 	"github.com/Azure/aks-app-routing-operator/pkg/controller/testutils"
-	"testing"
 
 	"github.com/go-logr/logr"
 	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+var restConfig *rest.Config
+
+func TestMain(m *testing.M) {
```

Comment:
Maybe the TestMain should be moved to a more generic file within this package since it's not strictly tied to the resource_reconciler. `main_test.go` or something similar

Reply:
Moved to `main_test.go`